contracts/impls/centralized/WitnetRequestBoard.sol

```diff
@@ -0,0 +1,319 @@
+import "../../utils/Destructible.sol";
+import "../../utils/Upgradable.sol";
+
+/**
+ * @title Witnet Requests Board mocked
+ * @notice Contract to bridge requests to Witnet for testing purposes.
+ * @dev This contract enables posting requests that Witnet bridges will insert into the Witnet network.
+ * The result of the requests will be posted back to this contract by the bridge nodes too.
+ * @author Witnet Foundation
+ */
+contract WitnetRequestBoard
+    is 
+        Destructible,
+        Upgradable, 
+        WitnetBoard, 
+        WitnetBoardDataACLs
+{
+    uint256 internal constant __ESTIMATED_REPORT_RESULT_GAS = 102496;
+
+    constructor(bool _isUpgradable) Upgradable(_isUpgradable) {}
+
+    // ================================================================================================================
+    // --- Overrides 'Destructible' -----------------------------------------------------------------------------------
+
+    function destroy() external override onlyOwner {
+        selfdestruct(payable(msg.sender));
+    }
+
+
+    // ================================================================================================================
+    // --- Overrides 'Upgradable' -------------------------------------------------------------------------------------
+
+    /// @dev Initialize storage-context when invoked as delegatecall. 
+    /// @dev Should fail when trying to initialize same instance more than once.
+    function initialize(bytes memory _initData) virtual external override {
+        address _owner = __data().owner;
+        if (_owner == address(0)) {
+            // set owner if none set yet
+            _owner = msg.sender;
+            __data().owner = _owner;
+        } else {
+            // only owner can initialize:
+            require(msg.sender == _owner, "WitnetRequestBoard: only owner");
+        }        
+
+        if (__data().base != address(0)) {
+            // current implementation cannot be initialized more than once:
+            require(__data().base != base(), "WitnetRequestBoard: already initialized");
+        }        
+        __data().base = base();
+
+        emit Initialized(msg.sender, base(), codehash(), version());
+
+        // Do actual base initialization:
+        setReporters(abi.decode(_initData, (address[])));
+    }
+
+    /// @dev Tells whether provided address could eventually upgrade the contract.
+    function isUpgradableFrom(address from) external view override returns (bool) {
+        address _owner = __data().owner;
+        return (
+            // false if the WRB is intrinsically not upgradable
+            isUpgradable() && (                
+                _owner == address(0) ||
+                _owner == from
+            )
+        );
+    }
+
+    /// @dev Retrieves named version of current implementation.
+    function version() virtual public pure override returns (string memory) {
+        return "WitnetRequestBoard-Centralized-v0.3.1.3";
```

Comment:
I dont know if it is possible, but can it be read from a specific field?. My point is that if we update the contract but we miss update this function, it could provide wrong information

Reply:
Yep, I agree.

Perhaps a better way to go, is to let the operator (aka: deployer) tag the new implementation that pretends to deploy and upgrade from the proxy. Moreover, as our migrations scripts run on javascript, they could even fetch the version tag from the `package.json`, for instance. 

A couple of things to have in mind, though: 

- Strings cannot (yet) be declared in Solidity as immutable fields, so it's necessary to change declaration of this method to:
  `function version() public pure override returns (bytes32)`. 
- Actual version tag should be provided on construction.